processing/src/main/java/org/apache/druid/guice/BuiltInTypesModule.java

```diff
@@ -53,14 +61,15 @@ public List<? extends Module> getJacksonModules()
   public void configure(Binder binder)
   {
     registerSerde();
-    // binding our side effect class to the lifecycle causes registerHandler to be called on service start, allowing
-    // use of the config to get the system default format version
-    LifecycleModule.register(binder, SideEffectHandlerRegisterer.class);
+    // binding our side effect classes to the lifecycle causes the initDimensionHandlerAndMvHandlingMode to be
+    // called on service start, allowing use of the config to get the system default format version and string multi
+    // value handling mode.
+    LifecycleModule.register(binder, SideEffectRegisterer.class);
   }
 
   @Provides
   @LazySingleton
-  public SideEffectHandlerRegisterer registerHandler(DefaultColumnFormatConfig formatsConfig)
+  public static SideEffectRegisterer initDimensionHandlerAndMvHandlingMode(DefaultColumnFormatConfig formatsConfig)
```

Comment:
why is this static?

Reply:
Ah, that's because there was a findbugs error in the build for the assignment of `STRING_MV_MODE` :
```
 'write to static field from instance method'
```

Reply:
updated it to instance method that invokes a static method to set `STRING_MV_MODE`